### ISSUE TITLE
10.11 Support?

### ISSUE THREAD

**Issue by sammcj** [vinc3m1/mac-pixel-clock-patch#12]

**Comment by Floris497** [comment]
I think this will work on 10.11 if you disable rootless with a bootarg, but what is the point when  switchresx does not support 10.11. If you post your md5 i will add it to a test branch. I testen it in beta 1 without much succes.

**Comment by sammcj** [comment]
SwitchresX does work in 10.11 though - I'm currently on Beta 2 about to upgrade to 3 later today.

```
samm-mbp ~ % md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
516bea2790cd6bfcb41959aa66122623
```

**Comment by Floris497** [comment]
I will edit it in a few days, im currently not at home nor have a computer with me. The easyest way is to replace one of the last md5's in de top of the script. One of the IOKIT ones. It will work straight away :) but will tell you your parching a different file. Which is okay. Good luck!

**Comment by sammcj** [comment]
Thanks :)

**Comment by jzelez** [comment]
Here is the md5 for 10.11 beta 3:

```
 md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
 15f9046ff25c807b7c76db8cdaf6ae4c
```

Thanks!

**Comment by sammcj** [comment]
Here is the md5 for 10.11 beta 5:

```
md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit                                                                                           git:master*
798a8e362e89e38e8f9cf6e59fda5184
```

Thanks,

**Comment by sammcj** [comment]
FYI for this to work on 10.11, you need to disable rootless (don't do this unless you understand the potential security implications):

```
sudo nvram boot-args="rootless=0"
```

**Comment by Floris497** [comment]
I had no success when i treid it with beta 1, but i hope this will work then, i'll add some md5's when i get back home in a few hours

**Comment by Floris497** [comment]
I just added 10.11 support, make sure to disable rootless in beta 1-3 with the boot arg "rootless=0" and in later versions you can do it in the recovery partition under security settings.

also make sure you use the beta of SwitchResX if you use that.

This tool a little longer because i needed to make a new patch, the previous one has been the same since 10.9.3.

can someone provide me with a md5 of Beta 4? (if anyone is still running that)

Good Luck!

**Comment by sammcj** [comment]
```
samm-mb ~ % md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
798a8e362e89e38e8f9cf6e59fda5184
```

Latest dev beta (5?) as of 8/8/2015

**Comment by Floris497** [comment]
EDIT: whoops made a little mistake, should work now, we are currently on beta 6 as far as I know :)

**Comment by kathampy** [comment]
I am able to run a 538MHz pixel clock (2560x1440@144 Hz) without any patch on the 10.11 betas on my Mid 2013 rMBP.

**Comment by Floris497** [comment]
Hm, and 4k?

**Comment by kathampy** [comment]
I doubt it cares about the resolution. They may have unlocked the pixel clock on the nVidia GPU alone on some models. I don't know if the Intel-only ones work.

**Comment by sammcj** [comment]
I find that 4k is limited to 30hz on any device 2+ years old.

**Comment by Floris497** [comment]
Ah, this patch does not work on the latest nvidia/amd drivers. And is only updated for integrated graphics.

**Comment by Floris497** [comment]
My 2011 mini can get 4k to 40hz over DP

**Comment by adeze** [comment]
GM Candidate has been released..
md5 seems to be 

  cd40217cd8d2ed8f16fa4ca513253109

**Comment by rwkyyy** [comment]
I just updated to 10.11 aaannnddd after a lot of tries, nothing works :( I'll try now with SwitchResX, before, just this patch and it was enough.

(MacBook Air early 2015 - output to a 2560 x 1080 display)

**Comment by Floris497** [comment]
That should not be an issue at all, what cable do you use? HDMI should be able to do that resolution at 60Hz (or 50/55) did you use the latest one, i updated it a few times today

**Comment by rwkyyy** [comment]
@Floris497 well, on 10.10 with this patch I was able to use the external display at its highest resolution, now I'm not able to, anymore, if I won't find a solution I will downgrade to 10.10.

My setup is this one -> <a href="http://www.sandberg.it/en-us/product/Adapter-Mini-DisplayPort-HDMI">this</a> adapter connected to <a href="http://www.lg.com/us/monitors/lg-25UM57-P-ultrawide-led-monitor">this</a> monitor.

Without this patch I wasn't able to use the native resolution of the monitor, after the patch it worked flawless at 2560 x 1080 (my friend has the same setup only that he has 10.10, and an Air late 2013).

**Comment by Floris497** [comment]
Are you sure the patch worked, and that you disabled System Integrity Protection (aka rootless)? The script will give some errors if it is still enabled.

**Comment by rwkyyy** [comment]
yes, before the patch, I tried switchresx, and I checked on my monitor, it showed 1920 x 1080 @ 60, after the patch it show 2560 x 1080 @ 60hz, I enabled at 10.11 rootless, never before.

wierd thing tho, the terminal wouldn't give me any message while executing the patch (before and after rootless)

**Comment by Floris497** [comment]
Can you give me your current md5 hash? if the patch already ran the patch should work, you can try and place the old file back and try and run the patch again, when placing the file back use the mv command like this: $ mv file.bak file

**Comment by rwkyyy** [comment]
@Floris497 Good morning, sorry, didn't see your message.
This is my md5 - 131978134faf623c7803458c2a204d60
I haven't done any file modifications, just applied the patch.

**Comment by Floris497** [comment]
Have you applied the latest patch from my fork? I updated it a few times yesterday.. 😁

**Comment by rwkyyy** [comment]
@Floris497 I'm sorry, I can't find it, not a master of Github :( If I press Fork it will create a copy of your git.

<a href="http://imgur.com/33aNTgp">This</a> is how it shows in my Terminal when executing.

The latest update for the commit on Jul 25, 2014.

**Comment by Floris497** [comment]
https://raw.githubusercontent.com/Floris497/mac-pixel-clock-patch/master/macPixelClockPatcher.command here you go :D

**Comment by rwkyyy** [comment]
@Floris497 thank you, I'll try it on monday, I'm out of town for the weekend, will post back the results, thank you!

**Comment by Floris497** [comment]
@rwkyyy okay, cool hope it works!

**Comment by rwkyyy** [comment]
This is the result, surprising, this time it asked for admin password.

<blockquote>Can't rename /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit to /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.bak: Operation not permitted, skipping file.
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: Operation not permitted
Detected unpatched NVIDIA driver on 10.11, patching.
Can't rename /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal to /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal.bak: Operation not permitted, skipping file.
</blockquote>

I haven't tested it, yet, will come back with a response, but it shows there is a NVIDIA driver unpatched.
Also it displays a lot of operations not permitted, but when it asked for the password, I gave it to him, do I have to run it under root?

**Comment by Floris497** [comment]
No even root does not have the privilege to change that, it is the new kernel level security provided by 10.11, you need to disable it temporarily to change the IOKit, and completely to run the now unsigned Nvidia kext.. I had not forseen this problem, so i will change it soon, so it is optional to do the nvidia one.

**Comment by rwkyyy** [comment]
OK, so what should I do (still haven't tested if it worked).

I have to disable the security with rootless or something more? And after I do the patch re-enable it?

**Comment by Floris497** [comment]
Wait a few minutes, i will make NVidia update a choice, if you do not have an NVidia card you are good to go, you can disable SIP run the script, maybe edit some resolutions and enable it again. After reanableing it your NVidia kext wont work anymore, because SIP, also blocks unsigned kext files. You can also delete the Nvidia MD5's so the script won't detect them!

**Comment by rwkyyy** [comment]
OK, I will test it when I get to the office (in about 2 hours from now). I'll tell you if it worked or not.

**Comment by rwkyyy** [comment]
Not working :(

**Comment by Floris497** [comment]
What did you try last?

**Comment by Floris497** [comment]
@rwkyyy can't help you if i don't know what you did.

**Comment by rwkyyy** [comment]
@Floris497 I just tried to apply the patch, same error (used the link above).

**Comment by Floris497** [comment]
@rwkyyy make sure you disable SIP, this error should not occur when it is disabled, run `csrutil status` to see if it really is disabled

**Comment by rwkyyy** [comment]
@Floris497 Thank you, it worked, I had to disable SIP.

<blockquote>Detected unpatched IOKit on 10.11, patching.<br>
Password:<br>
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature<br>
Detected unpatched NVIDIA driver on 10.11, patching.</blockquote>

You can check <a href="http://i.imgur.com/5uFzaqy.png">this link</a> to see the resolution I'm running on, atm.

Thank you again!

**Comment by Floris497** [comment]
That looks great! Have fun with it! If you enable SIP again the Nvidia driver will not work anymore. The Intel graphics driver will work. This is because the nvidia kext gas been changed and a changes kext will not load with SIP.

**Comment by rwkyyy** [comment]
@Floris497 Thank you for the heads-up, I re-enabled SIP and it's OK.

**Comment by andrecsa** [comment]
Hi there, I own a MacBook Pro 15" (non-retina) early 2011 model and just bought an ASUS VX24AH that supports 2560x1440.
My problem is that the monitor supports between 50Hz and 70Hz vertical, and for the pixel clock to be under 165MHz, I have to set 42Hz of vertical scan rate on SwitchResX and then the monitor shows an OUT OF RANGE message... And if I go over 165MHz of pixel clock, SwitchResX shows "Not Activated - Invalid?" on the custom resolution.
So, tried applying the patch, but I'm still not able to go over 165MHz, same message as before...
Has anyone had a similar experience that could give me a hand?
Could the fact that my MacBook Pro has an AMD card be a problem?

My setup:
MacBook Pro 15" (early 2011) - AMD Radeon HD 6750M and Intel HD Graphics 3000.
OS X El Capitan (I noticed that @Floris497 has updated the patch for OS X 10.11.1, which is the version I have installed)
Moshi Mini DisplayPort to HDMI adapter (4K) - It's supposed to support up to 4K
Monitor Asus VX24AH (this version does not have DisplayPort, only 2 HDMIs)

(Attached is the SwitchResX setup I'm trying)
![not activated - invalid](https://cloud.githubusercontent.com/assets/15334159/10752124/119a6fa4-7cd2-11e5-9341-7ff08fb7c043.png)

**Comment by Floris497** [comment]
Did the script work? And do you have SIP disabled? Maybe your dedicated GPU kicks in when you connect the display.. My mini server from 2011 can handle 4k@40Hz via DP.. Look in the script and run the md5 command for the nvidia kext, then add that md5 to the script.. Should work. I can help you in some time, but not now.. Good luck

**Comment by andrecsa** [comment]
Yes, I did disabled SIP and the script worked, I even compared the disassembled versions of the original and patched IOKits and it looked correct as per your instructions.
I've got gfxCardStatus which is a little app to manage manually the two GPUs on the MacBook Pro, when I have the monitor connected, and I try to set manually to use the integrated one, it doesn't let me:
![screen shot 2015-10-27 at 7 26 37 pm](https://cloud.githubusercontent.com/assets/15334159/10754042/b20dd756-7ce0-11e5-9e61-174b00ada3b1.png)
And that's the reason why: "This is because if you were to do this, your discrete GPU would actually stay powered on, even though you've switched to the integrated GPU."
I'll try to do what you've suggested. I'll let you know how I go.
Thanks a lot.

**Comment by andrecsa** [comment]
So, just did what you suggested, got the md5 of the AMD, added to the script and ran it again. Apparently it was patched:
![screen shot 2015-10-27 at 7 46 22 pm](https://cloud.githubusercontent.com/assets/15334159/10754456/6de41092-7ce3-11e5-8d71-a2ed91ba1dee.png)
But still the same. Can't create resolutions with pixel clock over 165 in SwitchResX...
Any clue?
Thanks again.

**Comment by andrecsa** [comment]
One more piece for the puzzle: The only difference between the original AMD kext and the patched one is the first line (probably because I renamed the file before disassembling it). So I don't think it's been patched correctly...

![screen shot 2015-10-27 at 8 55 15 pm](https://cloud.githubusercontent.com/assets/15334159/10756049/3b2f4d92-7ced-11e5-8d06-46fdcb36db6c.png)

**Comment by Floris497** [comment]
The difference in file name is because otool puts it there when you disassemble the file.. Not sure if te AMD one is still up to date.. I will look into it when i get home.. You can look in the executible in xcode in hex view and cange every place your find 165MHz in hex and iirc it is also reversed that is how x86 asm works..

**Comment by andrecsa** [comment]
Thanks @Floris497. I'll give that a go but I think it's a little bit over my knowledge.

If you could give me a hand when you have a chance I'd appreciate it.

**Comment by Floris497** [comment]
there are a few things of which im nit sure you are doing them right, 1 did you change an MD5 of the AMD patch? And not the patched md5 to check? Does the script tell you anything? Is there a bak file? If you open the executible in Xcode and get the sidebar to appear, right click it and select something to open as hex, it is in a sub menu, see if you can find the hex code from the amd patch in the file.. If so the script should work.. You can also check if you can find 165000 in the script, if that would be xAB xCD xEF you have to convert it to xEF xCD xAB IIRC, use wolframalpha to see what it is, or some other tool, i can try and do this for you when i have time.. I know you probably want to fix this asap, so i hope this info will get there. I will see if i can help you tomorrow..

**Comment by andrecsa** [comment]
What I did was to add this line at the top of the script (which is the md5 of the original /System/Library/Extensions/AMDSupport.kext/Contents/MacOS/AMDSupport I have in my computer):

```
amd_md5_10_11_1=938e88a65f42e385110fa2a0d5716013
```

And at the end of the script, I added this block of code:

```
if [ "$amd_md5_current" = "$amd_md5_10_11_1" ]; then
   echo "Detected unpatched AMD driver on 10.11.1, patching."
   sudo perl -i.bak -pe '$oldLimit1 = qr"\x75\x0C\x49\x81\x7E\x28\x40\xB3\xD5\x09"s;$newLimit1 = "\x75\x0C\x49\x81\x7E\x28\x00\x84\xD7\x17";$oldLimit2 = qr"\xFF\xFF\x48\x81\x7D\x80\x41\xB3\xD5\x09"s;$newLimit2 = "\xFF\xFF\x48\x81\x7D\x80\x01\x84\xD7\x17";s/$oldLimit1/$newLimit1/g;s/$oldLimit2/$newLimit2/g' /System/Library/Extensions/AMDSupport.kext/Contents/MacOS/AMDSupport
   sudo touch /System/Library/Extensions
fi
```

Which I copied from another AMD version you had in the script and just replaced by my new md5.
Yes, a bak file was generated, but there were no differences between the bak and the patched kexts.

I opened the executable in Xcode as hex, and I could not find neither of the oldLimit/newLimit hex sequences (from the above code) in the file. That means that the original AMDSupport doesn't have the sequences defined in oldLimit1 or oldLimit2... 

I'll give a try with the new tips you gave.

Thanks.

**Comment by andrecsa** [comment]
What number should I replace 165000 for? Any number above the pixel clock I need?

**Comment by andrecsa** [comment]
This is what I came up with this time:

Found only one instance of 165000000 (x40\xB3\xD5\x09) in the AMDSupport binary and replaced it by 400000000 (x00\x84\xD7\x17). Did this by modifying your script:

```
if [ "$amd_md5_current" = "$amd_md5_10_11_1" ]; then
    echo "Detected unpatched AMD driver on 10.11.1, patching."
    sudo perl -i.bak -pe '$oldLimit1 = qr"\x85\x17\x00\x00\x00\xB8\x40\xB3\xD5\x09"s;$newLimit1 = "\x85\x17\x00\x00\x00\xB8\x00\x84\xD7\x17";s/$oldLimit1/$newLimit1/g' /System/Library/Extensions/AMDSupport.kext/Contents/MacOS/AMDSupport
    sudo touch /System/Library/Extensions
fi
```

This sequence "\x85\x17\x00\x00\x00\xB8\" is what preceded the 165000000, so I used it to make sure I was replacing the right thing.

This is the disassembled AMDSupport.bak (original) and AMDSupport (patched):
![screen shot 2015-10-28 at 8 09 42 pm](https://cloud.githubusercontent.com/assets/15334159/10785434/eacf10b2-7daf-11e5-9661-3978478f2f70.png)
So apparently the patch worked correctly...

But still no able to set the resolution I need.

**Comment by Floris497** [comment]
There are 2 limits in the older patch.. Think there are 2 in the new one as well.. It indeed sets the new limit to 400MHz i will try it later today, but i don't have an AMD GPU installed in my system right now.. So i cant test it..

**Comment by andrecsa** [comment]
Yeah, but I could not find the other limit (which is 165000001) in the AMDSupport I have in my computer, so I didn't bother adding to the script...
I'm happy to test if you come up with something.
Thanks.

**Comment by andrecsa** [comment]
Even though my MacBook Pro doesn't have NVidia, I've patched the NVDAGK100Hal and NVDAGF100Hal replacing every instance of 165000/165000000/165000001 that I found as well...

Still no luck. :-(

I think I'm out of ideas now.

Let me know if you have any other tip.

Thanks.

**Comment by Floris497** [comment]
I have no idea too... i never made the initial patch for AMD, and a recreate of the NVIDIA patch worked.. there is no documentation on this.. and unfortunately i don't have the hardware to test it.. maybe you can patch the AMD frame buffer, but don't ask me how.. maybe if you get an active DP to HDMI adapter.. i can't tell you what you need.. but 2560x1440 should be supported also if it exceeds the pixel clock.. and at least at 50Hz and 60Hz should not be a problem also..

**Comment by andrecsa** [comment]
Ok. I guess I'll have to live with 1080p... :-(

Thanks for helping anyway.

**Comment by andrecsa** [comment]
@Floris497, do you know who made the initial AMDSupport patch?? I've googled it, but could not find...

**Comment by Floris497** [comment]
No i have no idea, maybe your can find something in the google code project. Where this originally came from.. But having the right adapter is key! You should read some of the issue here and on the project i forked this from, and the comments on the Google project, there your can find what adapter you would need for this to work, this might even work without this patch if you have the right one!

**Comment by andrecsa** [comment]
I thought I didn't need an adapter to get the config working on SwitchResX... I thought it was a software thing not hardware...

For instance, it works fine when I boot my Mac on Windows (bootcamp). But I really, really wanted not to have to use Windows for this...

Thanks @Floris!!!

**Comment by andrecsa** [comment]
Hey @Floris497, just for your info, I've bought the active Mini DisplayPort to HDMI adapter from Amazon (http://www.amazon.com/Accell-B086B-008B-2-UltraAV-DisplayPort-Adapter/dp/B00DOZHL82/ref=sr_1_1?ie=UTF8&qid=1447221123&sr=8-1&keywords=accell+mini+displayport+to+hdmi+adapter) and it worked out of the box, without patching.

**Comment by IsidoreIsou** [comment]
Hello
I tried to apply the patch on my Mac Mini 2012 on 10.11.1 (HD4000 > HDMI cable > AOC 28" UHD).
SIP was disabled. The patch worked without error.
But after that I only got a black screen on HDMI, and some glitch on DP in an other screen, the menu bar was strange, etc.
On SwitchResX, I can't run my custom rez, don't know why, it stays on 'Not installed'.
I did a Time Machine recovery to get back to normal.
What did happened ? Any advice ? What can I do to disable the patch without having to recovery ?
THANKS

**Comment by IsidoreIsou** [comment]
Actually... I DID disable SIP on recovery mode, I just did, after rebooting when I terminal csrutil status it says : System Integrity Protection status: enabled (Custom Configuration).
Anybody knows why can't I disable SIP ?
Many thanks..

**Comment by IsidoreIsou** [comment]
Hi
Sorry for all the edits. 
I use a 2012 Mac mini with a uhd screen connected through hdmi (NO adapt)
Everything works for me till i reboot and choose the uhd res on switchresx or monitor pref 
Then everything turn black and i can't do anything, even after a reboot my screen stays black. 
I did turn off SIP.
I fell a bit lost now
Thanks for your work anyway!

**Comment by Floris497** [comment]
I have had the same problem with the same display, I bought it a year ago and i got my replacement today, i just didn't feel like replacing it.. You have to unplug the power cable and make sure there is no power left in the screen by turning it of and on, if possible.. And try disconnecting and connecting the cable, and try DP, 2012 models seem to have problems with 4k with more recent systems.. Tell me if this works :) (you have a U2868PQU right?) if it stays black and gives little almost unhearable clicks it's trying to get the backlight on.. My problem with the display was that it sometimes flashed and showed a line in the middle, if you have that, you should ask for a replacement it!

**Comment by IsidoreIsou** [comment]
Wow thanks for those tips I Will try it soon
I do have this display. 
So your new model works better now ?
Thanks a lot

**Comment by Floris497** [comment]
As far as I know, just have it for a few hours, the problems don't show always, mostly after a few hours

**Comment by IsidoreIsou** [comment]
Here what I did :
- Apply the patch, reboot
- In monitor pref, change res to 1080p : Screen turn black with a 'little almost unbearable clicks'. Disconnect power cable, HDMI cable, nothing works.
- I connect the mac mini to an other monitor (U2412M) in DP : in works.
- Disconnect HDMI and connect in DP to the AOC : screen goes on, looks like a retina 1080p mode, with the menu bar buggy. I try to change the res to default (UHD maybe?) : goes black again.
- Can't do anything from here, stays black...

Doesn't seem to accept UHD res.. what do you think ?
Do I need to get an active adaptator ?  

Thanks a lot

**Comment by Floris497** [comment]
I'll see what i can find out, but I'm afraid you have a malfunctioning display too.. There have been batches of bad displays i think.. But most were recalled before they were sold.. Im not sure though, but you certainly have when you see a 1px wide line in the middle vertically, this vertical line is a copy of the vertical line at about 700px from the left side. But your computer most likely doesn't support 4k out of the box 😁

**Comment by IsidoreIsou** [comment]
Thanks for your help
I don't see the vertical line though
I am wondering of bying the Accell converter would solve it?

**Comment by Floris497** [comment]
I have no experience with an active converter like that or any other active converter.. I don't know the price.. Currently I'm trying to edit the framebuffer, but it looks like the script doesn't work for 10.11 anymore 🙈.. So I have to figure out how that works.. The script is very well documented so it should certainly be possible 😃

**Comment by IsidoreIsou** [comment]
Would you have idea how I could erasr the modifications the script have done on recovery mode? I can't get back on desktop, screen stays black, But I can boot on recovery. 
THX

**Comment by Floris497** [comment]
Place the backup file made by the patch back, maybe the computer is stuck in an unsupported resolution. Try resetting nvram that may help :)

**Comment by IsidoreIsou** [comment]
Hey @andrecsa , could you confirm what res works with the adaptater?
Have you tried 3200x1800@60Hz ? Could you? Thanks very much!

**Comment by Floris497** [comment]
I got 4k at 20Hz working quite stable with franebuffer kext edits, im trying to find the best solution, but it looks like the script is broken for 10.11 and even newer versions of 10.10.. And im not sure now if i can fix the script

**Comment by IsidoreIsou** [comment]
Ok Floris thanks for that I'll try the adapter and let you know
Le dim. 6 déc. 2015 à 13:46, Floris Fredrikze notifications@github.com a
écrit :

**Comment by IsidoreIsou** [comment]
Does it makes any sense :
- I turn on my Mac mini with the patch applied, both screens stays black as I said, BUT,
- I try to remote access my Mac mini from my macbook air : I can see both screens on Screen Sharing on the macbook Air, **I can see the full UHD res is working on my AOC on Screen Sharing app**. So your patch IS working. But the AOC won't ever wake up.
- Conclusion, this AOC screen has issues, right? should I return it ? Others UHD screen would work better ?

Thanks

**Comment by Floris497** [comment]
You should unplug the power of the display, then push the button on the aoc screen to change input, this will make sure all the power is drained that is still present in the display.. This resets the display. Then replug it, and see if it works.. You can also try to connect the monitor to a different computer or different cable to get it working.. You have to figure out what works exactly, but it has some problems. Also make sure you set it to between 20 and 40Hz.. 60Hz will show up, but doesn't work! Hope this helps :)

**Comment by IsidoreIsou** [comment]
The Accell adapter makes no difference. The UHD res works but the AOC won't get backlight on UHD, it stays black, however the res is working and the screen is displaying it. :( 
I will contact support now. Thanks for you help.

**Comment by Floris497** [comment]
Does the screen work at any different resolution?

**Comment by IsidoreIsou** [comment]
It works at 1920x1080@60Hz (w/o Switchresx) and 2560x1440@60Hz (with
Switchresx)

On Switchresx,
3840 x 2160 @15Hz creates black screen but works,
3840 x 2160 @30Hz is "Refused by the system"

**Comment by Floris497** [comment]
Your display won't support frameratees under 20Hz i think, not sure, your system won't support 30Hz.. Find the google code repo of this there are instructions on how to edit your frambuffer kext
